utils/docker/Dockerfile.centos.7

```diff
@@ -151,9 +136,22 @@ RUN chown -R daos_server.daos_server /home/daos
 USER daos_server:daos_server
 RUN ln -s utils/sl site_scons
 
+# Control what to build.  By default Dockerfiles build everything to allow for
+# ease-of-use for users, however in CI everything is turned off and then
+# selectively enabled.  Turning off any step disables all future steps.
+ARG DAOS_DEPS_BUILD=yes
+ARG DEPS_JOBS=1
+
 RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || \
-    (scons --build-deps=yes install PREFIX=/opt/daos --deps-only && \
-    /bin/rm -rf build *.gz)
+    { scons --build-deps=yes --jobs $DEPS_JOBS PREFIX=/opt/daos --deps-only && \
+    /bin/rm -rf build *.gz; }
+
+USER root:root
+# force an upgrade to get any newly built RPMs
+ARG CACHEBUST
```

Comment:
What's the rationale for having a cache-bust here and not before building dependencies?  I understand why it's preferable to have it here -- so that it doesn't invalidate the dependency cache layer.  But if you are not going to build the dependencies with the latest packages, why even do it here?

Reply:
This will ensure that DAOS itself is built with the latest dependencies although I do see there's an argument for "why do it at all".  Rebuilding the dependencies should happen rarely so having it here does give you a big win.

I supposed what would be ideal would be if you could run an extra step conditional on if the image had changed to do an update as part of the build.  Let me have a think about this, as it might be possible to copy in the files, do the update, do a deps build, then do the cachebust and another update which would ensure that if deps were ever built they'd be done with the latest packages.

Reply:
I'm not sure if this is what you are describing above, but what is actually needed is to determine if the update actually did update anything and apply a cachebust (only) if it did.  I'm not sure if that's possible though.  For `yum` (DNF will be different of course) there is `/var/log/yum.log` which gets appended to when any packages are installed.

There is also `yum history` or `rpm -qa | sort` before and after the update, which you can compare the output of.

Reply:
Ok, done.  There's an extra "yum update" between the copy in of the dependencies builder code and actually building them, this means that if the deps are going to be rebuilt it'll always update the software first.